### PR TITLE
Feature/nmd 452 set up a generic service

### PR DESCRIPTION
**Pull request by MaliyOna**

### CODE REVIEW

File: CatNote.DAL/Repositories/Interfaces/IGenericRepository.cs [hunk 1]
```diff
@@ -4,9 +4,9 @@ namespace CatNote.DAL.Repositories.Interfaces;
 
 public interface IGenericRepository<TEntity> where TEntity : IBaseEntity
```
Comment (GodlyCat) [inline CatNote.DAL/Repositories/Interfaces/IGenericRepository.cs:5]: Move this folder to the root of project

File: CatNote.BLL/Services/IGenericService.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using CatNote.DAL.Entities.Abstractions;
+
+namespace CatNote.BLL.Services;
+
+public interface IGenericService<TModel>
```
Comment (GodlyCat) [inline CatNote.BLL/Services/IGenericService.cs:10]: Create folder for Interfaces and move this folder to the root of the project

File: CatNote.DAL/Repositories/GenericRepository.cs [hunk 3]
```diff
@@ -43,9 +45,11 @@ public async Task<TEntity> GetById(int id, CancellationToken cancellationToken)
         return result!;
```
Comment (GodlyCat) [inline CatNote.DAL/Repositories/GenericRepository.cs:45]: Add AsNoTracking to GetById also

File: CatNote.BLL/Services/GenericService.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    public async Task<TModel> Create(TModel model, CancellationToken cancellationToken)
+    {
+        var entity = _mapper.ToEntity(model);
+
+        var resultEntity = await _genericRepository.Create(entity, cancellationToken);
+
+        return _mapper.FromEntity(resultEntity);
+    }
+
+    public async Task Delete(int id, CancellationToken cancellationToken)
```
Comment (GodlyCat) [inline CatNote.BLL/Services/GenericService.cs:28]: Convert to expression body

**Review by GodlyCat — CHANGES_REQUESTED** [review]

**Review by Finka95 — APPROVED** [review]

**Review by GodlyCat — APPROVED** [review]

**Pull request merged**